server/imports/core/connection/index.js

```diff
@@ -279,105 +431,318 @@ Connection.prototype = {
+      if (connection.options.replicaSetName) {
+        connectionUrl += ConnectionHelper.addOptionToUrl(
+          connectionUrl,
+          'replicaSet',
+          connection.options.replicaSetName
+        );
+      }
     }
 
-    if (connection.ssl && connection.ssl.enabled) connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'ssl', 'true');
-    if (connection.authenticationType) connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authMechanism', connection.authenticationType.toUpperCase().replace(new RegExp('_', 'g'), '-'));
+    if (connection.ssl && connection.ssl.enabled) {
+      connectionUrl += ConnectionHelper.addOptionToUrl(
+        connectionUrl,
+        'ssl',
+        'true'
+      );
+    }
+    if (connection.authenticationType) {
+      connectionUrl += ConnectionHelper.addOptionToUrl(
+        connectionUrl,
+        'authMechanism',
+        connection.authenticationType
+          .toUpperCase()
+          .replace(new RegExp('_', 'g'), '-')
+      );
+    }
 
     if (addAuthSource) {
-      if (connection.authenticationType === 'mongodb_cr' || connection.authenticationType === 'scram_sha_1' || connection.authenticationType === 'scram_sha_256') {
-        if (connection[connection.authenticationType].authSource) connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authSource', connection[connection.authenticationType].authSource);
-        else connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authSource', connection.databaseName);
-      } else if (connection.authenticationType === 'gssapi' || connection.authenticationType === 'plain') {
-        connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authSource', '$external');
+      if (
+        connection.authenticationType === 'mongodb_cr' ||
+        connection.authenticationType === 'scram_sha_1' ||
+        connection.authenticationType === 'scram_sha_256'
+      ) {
+        if (connection[connection.authenticationType].authSource) {
+          connectionUrl += ConnectionHelper.addOptionToUrl(
+            connectionUrl,
+            'authSource',
+            connection[connection.authenticationType].authSource
+          );
+        } else {
+          connectionUrl += ConnectionHelper.addOptionToUrl(
+            connectionUrl,
+            'authSource',
+            connection.databaseName
+          );
+        }
+      } else if (
+        connection.authenticationType === 'gssapi' ||
+        connection.authenticationType === 'plain'
+      ) {
+        connectionUrl += ConnectionHelper.addOptionToUrl(
+          connectionUrl,
+          'authSource',
+          '$external'
+        );
       }
     }
 
     return connectionUrl;
   },
 
+  getShellConnectionParams(connectionUrl) {
+    Logger.debug({
+      message: 'getShellConnectionParams',
+      metadataToLog: {
+        connectionUrl,
+      },
+    });
+    let mongoShellParams = [connectionUrl];
```

Comment:
Codacy has a fix for the issue: ['mongoShellParams' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
    const mongoShellParams = [connectionUrl];
```